### DISCUSSION TITLE
[Bug report] Possible bug with {{#is "index"}} in themes

### DISCUSSION THREAD

**Discussion by gpsblues** [GetPublii/Publii#1772] [Bug Report] [answered]
### Operating system

Win 11

### Publii version

0.46.1 (build 16918)

### Issue type

Free theme

### Bug description

During discussion [#1764](https://github.com/GetPublii/Publii/discussions/1764#discussioncomment-11034112), I encountered a possible bug with the {{#is "index"}} helper in the Mercury theme. I couldn’t verify whether this behavior is common across other themes.

When setting a custom page as the frontpage, it is not recognized as "index," which results in the scripts that should be included on the homepage being ignored. For instance, in the Mercury theme, the Isotope library fails to load.

To resolve this issue, I added the argument "page" to the {{#is}} helper, as shown below:

```
{{#is "index,tag,tags,author,page"}} 
<script defer src="{{js "isotope.pkgd.min.js"}}"></script> 
... 
{{/is}}
```
This change ensured that the Isotope library loaded correctly on the homepage.

It might be beneficial to investigate this issue further or update the documentation for the {{#is}} helper to clarify its functionality with custom homepage settings.

Thank you

**Comment by bobmitro** [comment, marked as answer]
It works as it should. The index is the index, and a page is a page :) 
When you decide to display the **page** on the **homepage**, the **index** is used to show posts on the **post prefix page**, e.g., /blog/. 

Regarding the Mercury theme, the isotope script should not be displayed on the homepage, as a **page** is currently displayed there.

**Comment by gpsblues** [comment]
Thank you,

I thought Index was the home page in any case. Thanks again, you clarified
the functioning of {{is}}.



Il giorno ven 25 ott 2024 alle ore 17:04 Bob Mitro ***@***.***>
ha scritto:
